src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/Internal/Json/FieldMaskConverter.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Linq;
+using System.Text.Json;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Type = System.Type;
+
+namespace Microsoft.AspNetCore.Grpc.JsonTranscoding.Internal.Json;
+
+internal sealed class FieldMaskConverter<TMessage> : SettingsConverterBase<TMessage> where TMessage : IMessage, new()
+{
+    public FieldMaskConverter(JsonContext context) : base(context)
+    {
+    }
+
+    public override TMessage? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        var message = new TMessage();
+
+        if (reader.TokenType != JsonTokenType.String)
+        {
+            throw new InvalidOperationException("Expected string value for FieldMask.");
+        }
+        // TODO: Do we *want* to remove empty entries? Probably okay to treat "" as "no paths", but "foo,,bar"?
```

Comment:
If we didn't remove empty entries, "" would give a single path and "," would give two? I think removing empty entries probably makes sense, but I don't know the spec.

Reply:
This field mask logic matches what is in the Google.Protobuf serializer. I'll add to the todo that we should follow their lead.